### PR TITLE
VectorImpl documentation

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

This is the final set of changes from the VectorImpl line. This is entirely documentation additions to assist with the creation of new spectre vector types derived from `VectorImpl`. I've decided to put this PR following on `ComplexDataVector`, as there are a few examples that reference using complex values as a storage type, and a couple of macro adjustments made in that PR.

### Types of changes:

- [ ] New feature

### Component:

- [ ] Documentation

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Dependencies
- [x] #970

### CODE REVIEW

File: docs/DevGuide/ImplementingVectors.md [hunk 1]
```diff
@@ -62,25 +64,30 @@ Blaze library. See [blaze-wiki/Vector_Operations]
...
+header file associated with the `VectorImpl` specialization. For `DataVector`,
+the specializations are defined in `DataStructures/DataVector.hpp`.The presence
+or absence of template specializations of these structs also determines the set
+of allowed operations between the vector type and other types. For example, if
+adding a `double` to a `DataVector` should be allowed and the result should be
+treated as a `DataVector` for subsequent operations, then the struct
+`blaze::AddTrait<DataVector, double>` needs to be defined as follows:
+
+
+abce
```
Comment (nilsdeppe) [inline docs/DevGuide/ImplementingVectors.md]: I think you forgot to add an example here, or typed a bit too much :)

File: docs/DevGuide/ImplementingVectors.md [hunk 2]
```diff
@@ -62,25 +64,30 @@ Blaze library. See [blaze-wiki/Vector_Operations]
...
 Blaze keeps track of the return type of unary and binary operations using "type
-trait" structs. The presence or absence of template specializations of these
-structs also determines the set of allowed operations between the vector type
-and other types. For example, if adding a `double` to a `DataVector` should be
-allowed and the result should be treated as a `DataVector` for subsequent
-operations, then the struct `blaze::AddTrait<DataVector, double>` needs to be
-defined as follows:
+trait" structs. These specializations for vector types should be placed in the
+header file associated with the `VectorImpl` specialization. For `DataVector`,
+the specializations are defined in `DataStructures/DataVector.hpp`.The presence
```
Comment (nilsdeppe) [inline docs/DevGuide/ImplementingVectors.md]: [nitpick in case you care, I don't] there's a missing space before "The" at the end of this line

**Review by nilsdeppe — COMMENTED** [review]
A couple minor things, I'm happy with the changes.

@kidder @gsb76 @carmaza since you were all there at the discussion could you please check that your concerns were addressed? Thanks :)

**Comment by moxcodes** [comment]
Thanks for catching those typos @nilsdeppe! I've taken the liberty of just including the fixes in the existing fixup commit.

**Review by kidder — COMMENTED** [review]
I am happy for you to squash once @nilsdeppe agrees

**Comment by nilsdeppe** [comment]
Sounds good to me :+1:

**Comment by moxcodes** [comment]
Squashed.
Cheers!

**Review by nilsdeppe — DISMISSED** [review]

**Review by wthrowe — DISMISSED** [review]

**Review by nilsdeppe — DISMISSED** [review]

**Comment by moxcodes** [comment]
Rebased on develop, which eliminates the clang-tidy fix commit that was also changed in #1080.

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Comment by kidder** [comment]
travis failure is the segfault one

**Pull request merged**